api/app/Services/Weather/WeatherService.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace App\Services\Weather;
+
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Cache;
+use Carbon\Carbon;
+
+class WeatherService
+{
+
+    private const CACHE_TTL_SECONDS = 3600; //1hours
+
+    public function __construct(private WeatherClient $client) {}
+
+    //Try to pull from cache
+    //TODO: maybe wrap below in a try catch? so i can fall back to stale cache cleanly:
+
+    public function getCurrentForUser(int $userId, float $lat, float $long): array
+    {
+        $key = "user:{$userId}:weather:v1";
```

Comment:
👍🏻 I like that you put some thought into this cache key so that it's unique, scoped, and versioned.